streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java

```diff
@@ -113,7 +102,37 @@ public InternalStreamsBuilder(final InternalTopologyBuilder internalTopologyBuil
         internalTopologyBuilder.addProcessor(name, processorSupplier, source);
 
         final KTableImpl<K, ?, V> kTable = new KTableImpl<>(this, name, processorSupplier,
-            consumed.keySerde(), consumed.valueSerde(), Collections.singleton(source), storeSupplier.name(), isQueryable);
+            consumed.keySerde(), consumed.valueSerde(), Collections.singleton(source), storeBuilder.name(), isQueryable);
+
+        internalTopologyBuilder.addStateStore(storeBuilder, name);
+        internalTopologyBuilder.connectSourceStoreAndTopic(storeBuilder.name(), topic);
+
+        return kTable;
+    }
+
+    private <K, V> KTable<K, V> doTable(final ConsumedInternal<K, V> consumed,
```

Comment:
Could we refactor this two private functions a bit differently? What I had in mind is that we can have one private function that include 1) addSource, 2) addProcessor, 3) new KTableImpl, and in the two public `table` APIs we do 0) create processorSupplier, and call the private function, and then 4) addStateStore, 5) connectSource.

In this way the really common path 1/2/3) is consolidated in this private function.